### PR TITLE
add `configure_no_prefix` option to skip addition of prefix to configure command in ConfigureMake easyblock

### PR DESCRIPTION
**Pull request by jfgrimm**

### CODE REVIEW

File: easybuild/easyblocks/generic/configuremake.py [hunk 1]
```diff
@@ -184,6 +184,7 @@ def extra_options(extra_vars=None):
                                  " False implies to leave it up to the configure script)", CUSTOM],
             'configure_cmd': [DEFAULT_CONFIGURE_CMD, "Configure command to use", CUSTOM],
             'configure_cmd_prefix': ['', "Prefix to be glued before ./configure", CUSTOM],
+            'configure_no_prefix': [False, "Avoid adding a prefix to the configure command", CUSTOM],
```
Comment (branfosj) [inline easybuild/easyblocks/generic/configuremake.py:187]: The different uses of `prefix` in `extra_vars` is causing me confusion. I.e. this has no impact on `configure_cmd_prefix`. I think the following is clearer:

```
"Avoid passing an install directory (such as through --prefix) to the configure command"
```

Also if someone sets `configure_no_prefix = True` and specifies a `prefix_opt` we should warn or exit.

Comment (branfosj) [reply easybuild/easyblocks/generic/configuremake.py:187]: And if `configure_no_prefix` would be better as `configure_without_installdir`.

Comment (jfgrimm) [reply easybuild/easyblocks/generic/configuremake.py]: that said, it's still sensible. changes made in latest commit

File: easybuild/easyblocks/generic/configuremake.py [hunk 2]
```diff
@@ -303,11 +305,19 @@ def configure_step(self, cmd_prefix=''):
             if host_type:
                 build_and_host_options.append(' --host=' + host_type)
 
+        if self.cfg.get('configure_without_installdir'):
+            configure_prefix = ''
+            if self.cfg.get('prefix_opt'):
+                print_warning("Specified prefix_opt '%s' is ignored due to use of configure_without_installdir",
+                              prefix_opt)
```
Comment (akesandgren) [inline easybuild/easyblocks/generic/configuremake.py:312]: I wonder if this should rather be an error than just a warning.... opinions??

**Comment by branfosj** [comment]
Test report by @branfosj
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _JasPer-4.0.0-GCCcore-12.2.0.eb_ 
 * **SUCCESS** _expat-2.4.9-GCCcore-12.2.0.eb_ 
 * **SUCCESS** _ncurses-6.3-GCCcore-12.2.0.eb_ 

Build succeeded for 3 out of 3 (3 easyconfigs in total)
bear-pg0105u36b.bear.cluster - Linux RHEL 8.5, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/6f56091f84f565383380107e7a2b3ee8 for a full test report.

File: easybuild/easyblocks/generic/configuremake.py (hunk 2 above)
Comment (branfosj) [reply easybuild/easyblocks/generic/configuremake.py:312]: I'm happy with it as a warning.

**Review by branfosj — APPROVED** [review]

**Pull request merged**